perf_bench_test.go

```diff
@@ -655,30 +664,23 @@ func bindVMsToSocket() error {
 		return err
 	}
 
-	cpus, err := cpuInfo.SocketCPUs(*bindSocket)
-	if err != nil {
-		return err
-	}
-
-	subCPUs, err = getCoresExceptProfiled(cpuInfo, cpus)
-	if err != nil {
-		return err
+	if socket > -1 {
+		cpus, err = cpuInfo.SocketCPUs(socket)
+		if err != nil {
+			return err
+		}
+	} else {
+		cpus = cpuInfo.AllCPUs()
 	}
 
 	vmPidList := strings.Split(string(pidBytes), " ")
 	for _, vm := range vmPidList {
```

Comment:
Lines 677+ are unclear. Can you please add a comment that describes the algorithm? Particularly, what the purpose of the `coreFree` variable is (also, it should be defined right before its usage).

Reply:
I really don't get how this works

Reply:
It has redundant logic from previous updates. I have simplified the logic and added comments